icaruscode/Generators/HepMCFileGen_module.cc

```diff
@@ -0,0 +1,457 @@
+#include "larcore/Geometry/Geometry.h"
+#include "larcoreobj/SummaryData/RunData.h"
+#include "larcoreobj/SummaryData/POTSummary.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+#include "nusimdata/SimulationBase/MCParticle.h"
+#include "ifdh.h"       // use bare
+#undef USE_IFDH_SERVICE // ifdh for now
+// #ifndef NO_IFDH_LIB
+//   #define USE_IFDH_SERVICE 1
+//   // IFDHC
+//   #ifdef USE_IFDH_SERVICE
+//     #include "IFDH_service.h"
+//   #else
+//     // bare IFDHC
+//     #include "ifdh.h"
+//   #endif
+// #else
+//   #undef USE_IFDH_SERVICE
+//   // nothing doing ... use ifdef to hide any reference that might need header
+//   #include <cassert>
+// #endif
+namespace evgen {
+  class HepMCFileGen;
+}
+class evgen::HepMCFileGen : public art::EDProducer {
+public:
+  explicit HepMCFileGen(fhicl::ParameterSet const & p);
+  void produce(art::Event & e)        override;
+  void beginJob()                                 override;
+  void beginRun(art::Run & run)                   override;
+  void endSubRun(art::SubRun& sr)     override;
+private:
+//  void ExpandInputFilePatternsDirect();
+//  void ExpandInputFilePatternsIFDH();
+  void open_file();
+  std::ifstream* fInputFile;
+//  std::string              fFileSearchPaths; ///< colon separated set of path stems (to be set)
+//  std::vector<std::string> fFilePatterns;    ///< wildcard patterns files containing histograms or ntuples, or txt (to be set)
+//  std::vector<std::string> fSelectedFiles;   ///< flux files selected after wildcard expansion and subset selection
+//  std::string              fFileCopyMethod;  ///< "DIRECT" = old direct access method, otherwise = ifdh approach schema ("" okay)
+  
+  double         fEventsPerPOT;     ///< Number of events per POT (to be set)
+  int            fEventsPerSubRun;  ///< Keeps track of the number of processed events per subrun
+  ifdh_ns::ifdh* fIFDH;             ///< (optional) flux file handling
+};
+//------------------------------------------------------------------------------
+evgen::HepMCFileGen::HepMCFileGen(fhicl::ParameterSet const & p)
+  : EDProducer{p}
+  , fInputFile(0)
+//  , fFileSearchPaths{p.get<std::string>("FileSearchPaths")}
+//  , fFilePatterns{p.get<std::vector<std::string>>("FilePatterns")}
+//  , fFileCopyMethod{p.get<std::string>("FluxCopyMethod","DIRECT")}
+  , fEventsPerPOT{p.get<double>("EventsPerPOT", -1.)}
+  , fEventsPerSubRun(0)
+{
+  srand (time(0));
+  produces< std::vector<simb::MCTruth>   >();
+  produces< sumdata::RunData, art::InRun >();
+  produces< sumdata::POTSummary, art::InSubRun >();
+}
+//------------------------------------------------------------------------------
+
+//************** Animesh modified it to include file from ICARUS_data area ********************//
+
+void evgen::HepMCFileGen::open_file()
+{
+
+   std::string fullFileName;
+    cet::search_path searchPath("FW_SEARCH_PATH");
+    searchPath.find_file("Darkmatterfile/ldm-test.root_ldm.hepmc",fullFileName);
```

Comment:
I have a problem with this approach: we have a fixed set of randomly generated events, stored in a standard repository (`icarus_data`) and we have a module to process those events.
I would *greatly* prefer the input file path to be passed as a parameter of the module.

Also I discourage the naming of a file as something containing `.root` as some code will find `.root` in the name and assume this is a ROOT file. I _recommend_ changing that `.` into something else (underscore maybe `_`?)

Reply:
I can change the name of the file and remove .root extension. but i not sure I have understood the difference between calling the file from icarus_data area and passing the name as a parameter of the module, will that be much faster to access the file? Can you please explain little more?

Reply:
To have the file name as parameter (3):
```suggestion
    cet::search_path searchPath("FW_SEARCH_PATH");
    searchPath.find_file(fInputFilePath,fullFileName);
```
```suggestion
    cet::search_path searchPath("FW_SEARCH_PATH");
    searchPath.find_file("Darkmatterfile/ldm-test.root_ldm.hepmc",fullFileName);
```

Reply:
To have the file name as parameter (3):
```suggestion
    cet::search_path searchPath("FW_SEARCH_PATH");
    searchPath.find_file(fInputFilePath,fullFileName);
```